sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,1468 @@
+----
+
+This extension also introduces USM memory allocation functions with `properties` support that allow the USM memory allocation kind to be specified at runtime.
+In this case, the returned `annotated_ptr` will not have the `sycl::ext::oneapi::experimental::usm_kind` property (unless that property is also passed in).
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi::experimental;
+
+properties P4{bar, foo{1}};
+
+// APtr7 is of type annotated_ptr<int*, decltype(properties{bar})>
+auto APtr7 = annotated_malloc<int>(N, q, sycl::usm::alloc::device, P4);
+
+----
+
+If the USM memory allocation kind specified by a parameter to the allocation function is different than the USM memory allocation kind specified by the `sycl::ext::oneapi::experimental::usm_kind` property, the USM memory allocation kind specified by the property takes precedence.
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi::experimental;
+
+properties P6{usm_kind<sycl::usm::alloc::device>};
+
+// The memory allocated will be of kind sycl::usm::alloc::device
+auto APtr10 = annotated_malloc<int>(N, q, sycl::usm::alloc::host, P6);
+
+----
+
+
+If the `sycl::ext::oneapi::experimental::usm_kind` property specifies a different USM memory allocation kind than the function supports, the compiler will issue a diagnostic error.
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi::experimental;
+
+properties P7{usm_kind<sycl::usm::alloc::device>};
+
+auto APtr11 = annotated_malloc_host<int>(N, q, P7); // Error
+----
+
+The following example uses the compile-time-constant property `alignment`, defined in the link:../proposed/sycl_ext_oneapi_annotated_ptr.asciidoc[sycl_ext_oneapi_annotated_ptr] extension.
+When `alignment` is passed to a USM memory allocation function with `properties` support, it will appear on the returned `annotated_ptr` since it is a compile-time constant property.
+It also informs the runtime to allocate the memory with this alignment.
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi::experimental;
+
+properties P8{alignment<512>};
+// APtr12 is of type annotated_ptr<int, decltype(properties{alignment<512>, usm_kind<sycl::usm::alloc::device>})>
+// The raw pointer of APtr12 is aligned to 512 bytes
+auto APtr12 = annotated_malloc_device<int>(N, q, P8);
+
+properties P9{alignment<1>};
+// Allocate N integers.
+// Alignment must be at least sizeof(int) bytes.
+auto APtr13 = annotated_malloc_device<int>(N, q, P9); // Error
+
+properties P10{alignment<64>};
+// Allocate 512 bytes.
+// Alignment must be at least 512 bytes.
+auto APtr14 = annotated_malloc_device(512, q, P10); // Exception
+----
+
+This extension also introduces USM memory allocation functions with `properties` support that allow alignment to be specified at runtime, using a separate parameter of type `size_t`.
+If the compile-time constant `alignment` property is also passed in, it takes precedence over the alignment specified by the parameter of type `size_t`.
```

Comment:
This doesn't seem user friendly.  I would expect the resulting pointer to have both specified alignments.  I.e. the resulting pointer value should be multiple of the lowest common multiple of the two specified alignments.  Assuming both specified alignments are powers of two, this means that the larger of the two takes precedence.

Reply:
I have made this change. I've removed all exceptions and errors that were caused by differing alignments.